system/jlib/jlog.ipp

```diff
@@ -767,7 +767,7 @@ public:
     bool                      isActiveMonitor(const ILogMsgHandler * handler) const { return (find(handler) != NotFound); }
     bool                      changeMonitorFilter(const ILogMsgHandler * handler, ILogMsgFilter * newFilter);
     bool                      changeMonitorFilterOwn(const ILogMsgHandler * handler, ILogMsgFilter * newFilter) { bool ret = changeMonitorFilter(handler, newFilter); newFilter->Release(); return ret; }
-    LogMsgId                  getNextID() { return static_cast<LogMsgId>(atomic_add_exchange(&nextID, 1)); }
+    LogMsgId                  getNextID() { return nextID.fetch_add(1, std::memory_order_relaxed); }
```

Comment:
The increment is guaranteed to be atomic, and since it always returns the previous value - they should be guaranteed to be unique.

@mckellyln should give a second opinion to confirm my understanding.